### ISSUE TITLE
How to add global representations into MessagePassing

### ISSUE THREAD

**Issue by lfangyu09** [pyg-team/pytorch_geometric#7758]
### 🐛 Describe the bug

Hi,

Thanks a lot for providing this library. I want to add a global representation (see the below image) into `MessagePassing`, but I don't find a way to add it. [The tutorial](https://pytorch-geometric.readthedocs.io/en/latest/tutorial/create_gnn.html) only mentions node features and edge features. How can I add a global representation into `MessagePassing`?

![image](https://github.com/pyg-team/pytorch_geometric/assets/62622741/21832057-e749-46a8-aacd-919aa5f48d95)

### Environment

* PyG version:
* PyTorch version:
* OS:
* Python version:
* CUDA/cuDNN version:
* How you installed PyTorch and PyG (`conda`, `pip`, source):
* Any other relevant information (*e.g.*, version of `torch-scatter`):

**Comment by lfangyu09** [comment]
Thanks a lot for your reply.

I used to use `MetaLayer` to do it. But I still want to use `MessagePassing` to do it. Here is the code of `Interaction Network`. How can I add global representation to this network? Thanks.

`class InteractionNetwork(pyg.nn.MessagePassing):
    def __init__(self, hidden_size, layers):
        super().__init__()
        self.lin_edge = MLP(hidden_size * 3, hidden_size, hidden_size, layers)    ### after prorcessing, both # of node features and edge features become hidden size
        self.lin_node = MLP(hidden_size * 2, hidden_size, hidden_size, layers)

    def forward(self, x, edge_index, edge_feature):
        edge_out, aggr = self.propagate(edge_index, x=(x, x), edge_feature=edge_feature)
        node_out = self.lin_node(torch.cat((x, aggr), dim=-1))
        edge_out = edge_feature + edge_out
        node_out = x + node_out
        return node_out, edge_out

    def message(self, x_i, x_j, edge_feature):
        x = torch.cat((x_i, x_j, edge_feature), dim=-1)
        x = self.lin_edge(x)
        return x

    def aggregate(self, inputs, index, dim_size=None):
        out = torch_scatter.scatter(inputs, index, dim=self.node_dim, dim_size=dim_size, reduce="sum")
        return (inputs, out)`

**Comment by rusty1s** [comment]
For global features, I suggest to map them to node-level space if you want to combine them into your node representation, or edge-level space if you want to make use of them in `message`, e.g.:
```python
u = ...  # graph vector of shape [batch_size, num_features]
u_node = u[batch]  # [num_nodes, num_features]
u_edge = u[batch[edge_index[0]]  # [num_edges, num_features]
```
